cpp/src/arrow/util/config.h.cmake

```diff
@@ -45,6 +45,7 @@
 #cmakedefine ARROW_IPC
 #cmakedefine ARROW_JSON
 
+#cmakedefine ARROW_GCS
```

Comment:
Could you also update document for `FileSystemFromUri`like https://github.com/apache/arrow/pull/12932/files#diff-82a2d9a8424dfb436344c0802dc36275cd0fb87e2e9eb6399745fba43799f7e5 ?

Reply:
done.